src/app/Library/CrudPanel/CrudPanel.php

```diff
@@ -362,20 +370,26 @@ private function getRelationModelInstances($model, $relationString)
         $relationArray = explode('.', $relationString);
```

Comment:
Took me a while to understand what these methods do from the name. I don't think I did a good job reviewing & merging these 2 years ago, but we can fix that now 😄 

I'd rename:
- ```getRelationModelInstances()``` to ```getRelatedEntries()```;
- ```getModelAttributeFromRelation()``` to ```getRelatedEntriesAttributes()```;

Please correct me if this isn't what the methods do, or you don't think these names are better. If you're worried about Backwards compatibility, we can use the above as the _real_ names, and the _old_ ones as aliases.

Reply:
Indeed. That's what those methods are doing. I don't think many people, or any at all, had overridden those methods, or the methods that calls them, they are like dark functions, I also took a while figuring what we get and what we expect from them. 

It might work ok, and if it worked until now we are good, but it just seems like .. hum.. : 

```php
foreach ($endModels as $model => $entries) {
            $modelKey = (new $model())->getKeyName();
            if (is_array($entries) && ! isset($entries[$attribute])) {
                foreach ($entries as $entry) {
                    $attributes[$entry[$modelKey]] = $entry[$attribute];
                }
            } elseif (is_array($entries) && isset($entries[$attribute])) {
                $attributes[$entries[$modelKey]] = $entries[$attribute];
            } elseif ($entries->{$attribute}) {
                $attributes[$entries->{$modelKey}] = $entries->{$attribute};
            }
        }
```